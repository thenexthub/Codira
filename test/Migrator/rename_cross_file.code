// REQUIRES: objc_interop
// RUN: %empty-directory(%t.mod) && %target-language-frontend -emit-module -o %t.mod/Cities.codemodule %S/Inputs/Cities.code -module-name Cities -parse-as-library
// RUN: %empty-directory(%t) && %target-language-frontend -c -update-code -primary-file %s %S/Inputs/rename_cross_file_2.code -F %S/mock-sdk -I %t.mod -api-diff-data-file %S/Inputs/API.json -emit-migrated-file-path %t/rename_cross_file.code.result -emit-remap-file-path %t/rename_cross_file.code.remap -o /dev/null
// RUN: diff -u %S/rename_cross_file.code.expected %t/rename_cross_file.code.result

import Cities

extension FontWeight {
  init(x: Int, y: Int) { fatalError() }
  init(x: Int, y: Int, _ z: Int) { fatalError() }
}

fn foo() {
  _ = FontWeight(rawValue: 1)
  _ = FontWeight(string: "light")
  _ = FontWeight(x: 2, y: 2, z: 2)
  _ = FontWeight(x: 2, 2)
}
