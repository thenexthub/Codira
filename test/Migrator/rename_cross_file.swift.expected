// REQUIRES: objc_interop
// RUN: %empty-directory(%t.mod) && %target-language-frontend -emit-module -o %t.mod/Cities.codemodule %S/Inputs/Cities.code -module-name Cities -parse-as-library
// RUN: %empty-directory(%t) && %target-language-frontend -c -update-code -primary-file %s %S/Inputs/rename_cross_file_2.code -F %S/mock-sdk -I %t.mod -api-diff-data-file %S/Inputs/API.json -emit-migrated-file-path %t/rename_cross_file.code.result -emit-remap-file-path %t/rename_cross_file.code.remap -o /dev/null
// RUN: diff -u %S/rename_cross_file.code.expected %t/rename_cross_file.code.result

import Cities

extension FontWeighting {
  init(x: Int, y: Int) { fatalError() }
  init(x: Int, y: Int, _ z: Int) { fatalError() }
}

fn foo() {
  _ = FontWeighting(rawValue: 1)
  _ = FontWeighting(string: "light")
  _ = FontWeighting(x: 2, y: 2, z: 2)
  _ = FontWeighting(x: 2, 2)
}
